### PR TITLE
Feat(build): Raknet as a first class citizen

### PR DESCRIPTION
**Pull request by magicaldave**
The goal of this PR is to better integrate Raknet into the build process so nobody has to mess with it anymore and make a couple other small tweaks to optimize builds.

- Adds raknet as a repo submodule
- Locates and initializes the repo through cmake
- Builds it and links using the existing mechanics in the other CMakeLists.

~~There is also a potentially controversial change in that I have made all outputs go directly to a folder, `build`. I could be approaching the problem incorrectly, but I believe outputting directly to build will make our job much easier for the purpose of generating releases.~~

~~Finally, this PR also disables building things like the ESS importer and BSA Tool by default, as nobody uses these and they're dramatically outdated in our repo anyway.~~

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,11 +23,11 @@ option(BUILD_OPENMW             "Build OpenMW" ON)
 option(BUILD_LAUNCHER           "Build Launcher" ON)
 option(BUILD_WIZARD             "Build Installation Wizard" ON)
 option(BUILD_MWINIIMPORTER      "Build MWiniImporter" ON)
-option(BUILD_OPENCS             "Build OpenMW Construction Set" ON)
-option(BUILD_ESSIMPORTER        "Build ESS (Morrowind save game) importer" ON)
-option(BUILD_BSATOOL            "Build BSA extractor" ON)
-option(BUILD_ESMTOOL            "Build ESM inspector" ON)
-option(BUILD_NIFTEST            "Build nif file tester" ON)
+option(BUILD_OPENCS             "Build OpenMW Construction Set" OFF)
```
Comment (SaintMercury) [inline CMakeLists.txt:26]: we shouldn't remove these or undefault them. Nor should we be breaking them for wahtever reason. The ESS I can see, but the others could be useful for server hosters or anyone w/ the engine.

Comment (magicaldave) [reply CMakeLists.txt:26]: The defaults should only affect us, if we use our existing CI arch then we can build these apps for releases without it slowing down our own builds; what do you think of that?

File: cmake/FindRakNet.cmake [hunk 2]
```diff
@@ -6,77 +6,58 @@
...
-  ENDIF(NOT RakNet_FIND_QUIETLY)
-ELSE(RakNet_FOUND)
-  IF(RakNet_FIND_REQUIRED)
-    MESSAGE(FATAL_ERROR "Could not find RakNet")
-  ENDIF(RakNet_FIND_REQUIRED)
-ENDIF(RakNet_FOUND)
+SET(RakNet_INCLUDES ${CMAKE_SOURCE_DIR}/extern/raknet/include/raknet)
 
+IF (CMAKE_CONFIGURATION_TYPES OR CMAKE_BUILD_TYPE)
+   SET(RakNet_LIBRARY optimized ${RakNet_LIBRARY_RELEASE} debug ${RakNet_LIBRARY_DEBUG})
```
[the remark this answers is not in the document]
Comment (magicaldave) [reply cmake/FindRakNet.cmake:52]: Definitely.

File: CMakeLists.txt (hunk 1 above)
Comment (magicaldave) [reply CMakeLists.txt:26]: Hmmm. If we do get breakages, we can let CI canary those (I imagine we're likely not using the unit tests locally either). Buuuutttt we can likely just get a little more comfortable with our generators if we really want the time saved.

**Comment by magicaldave** [comment]
Alright, after updating this, the Git is kind of messy and I can't exactly roll the changes back due to catching up to master.

So here's the deal. This MR *just* adds the new cmake for raknet and doesn't otherwise modify the build process. However, it's worth noting that the current job for this might fail due to a transient issue in github 22.04 runners. I'm opening a separate branch and MR to handle this and accomodate this.

For review purposes please refer to this job: https://github.com/DreamWeave-MP/Dreamweave/actions/runs/6516067790/job/17699187774

**Review by HotaruBlaze — APPROVED** [review]

File: cmake/FindRakNet.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (magicaldave) [reply cmake/FindRakNet.cmake:52]: I apologize, but on a quick look over this I can't really tell how we'd benefit from changing the branch logic. Per the 24 hour hold on PRs, We'll touch on this tomorrow and should be able to clear it up fairly quick!

**Pull request merged**